### PR TITLE
send AUTH message if sentinel password is set

### PR DESCRIPTION
**Pull request by kkolanow**
Hi, I've noticed that during connection to the password protected sentinel (assume that master, replica slaves and sentinels share same password) I was getting error NOAUTH Authorization required. I discovered that sentinel password was not passed on. I'm submitting a fix example . I've tried it and it worked for me.

### CODE REVIEW

**Comment by sazzad16** [comment]
@kkolanow There are several errors in `JedisSentinelPoolTest`. Please check the Travis CI log.

**Comment by kkolanow** [comment]
this time is fails on SetCommandsTest.  I'm afraid I'm not sure what  this test is supposed to do. Any help appreciated. Thanks!

File: src/main/java/redis/clients/jedis/JedisSentinelPool.java [hunk 1]
```diff
@@ -163,7 +191,7 @@ private HostAndPort initSentinels(Set<String> sentinels, final String masterName
       Jedis jedis = null;
       try {
         jedis = new Jedis(hap.getHost(), hap.getPort(), sentinelConnectionTimeout, sentinelSoTimeout);
-        if (sentinelPassword != null) {
+        if (sentinelPassword != null && !sentinelPassword.isEmpty())   {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisSentinelPool.java]: Check only `null`, not `isEmpty`.

File: src/main/java/redis/clients/jedis/JedisSentinelPool.java [hunk 2]
```diff
@@ -302,6 +330,9 @@ public void run() {
       while (running.get()) {
 
         j = new Jedis(host, port);
+        if (null != sentinelPassword && !sentinelPassword.isEmpty()) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisSentinelPool.java]: Check only `null`, not `isEmpty`.

File: src/main/java/redis/clients/jedis/JedisSentinelPool.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
 
   private volatile JedisFactory factory;
   private volatile HostAndPort currentHostMaster;
-  
+
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisSentinelPool.java]:
```suggestion
  
```

File: src/main/java/redis/clients/jedis/JedisSentinelPool.java [hunk 4]
```diff
@@ -163,7 +168,7 @@ private HostAndPort initSentinels(Set<String> sentinels, final String masterName
       Jedis jedis = null;
       try {
         jedis = new Jedis(hap.getHost(), hap.getPort(), sentinelConnectionTimeout, sentinelSoTimeout);
-        if (sentinelPassword != null) {
+        if (sentinelPassword != null)   {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisSentinelPool.java]:
```suggestion
        if (sentinelPassword != null) {
```

**Comment by sazzad16** [comment]
LGTM!

**Comment by gkorland** [comment]
Now that the ACL is in @kkolanow @sazzad16 I think the code should adapted accordingly

**Pull request merged**